src/frontend/src/lib/components/convert/ConvertAmountDestination.svelte

```diff
@@ -0,0 +1,67 @@
+<script lang="ts">
+	import { isNullish, nonNullish } from '@dfinity/utils';
+	import { getContext } from 'svelte';
+	import ConvertInputAmount from '$lib/components/convert/ConvertInputAmount.svelte';
+	import ConvertInputsContainer from '$lib/components/convert/ConvertInputsContainer.svelte';
+	import ConvertToken from '$lib/components/convert/ConvertToken.svelte';
+	import { ZERO } from '$lib/constants/app.constants';
+	import { CONVERT_CONTEXT_KEY, type ConvertContext } from '$lib/stores/convert.store';
+	import { i18n } from '$lib/stores/i18n.store';
+	import { formatToken, formatTokenBigintToNumber, formatUSD } from '$lib/utils/format.utils';
+	import { parseToken } from '$lib/utils/parse.utils';
+
+	export let sendAmount: number | undefined = undefined;
+	export let receiveAmount: number | undefined = undefined;
+	export let insufficientFunds: boolean;
+	export let totalFee: bigint | undefined = undefined;
+
+	const { destinationToken, destinationTokenBalance, destinationTokenExchangeRate } =
+		getContext<ConvertContext>(CONVERT_CONTEXT_KEY);
+
+	let sendAmountAfterFee: bigint | undefined;
+	$: sendAmountAfterFee =
+		nonNullish(totalFee) && nonNullish(sendAmount)
+			? parseToken({
+					value: `${sendAmount}`,
+					unitName: $destinationToken.decimals
+				}).toBigInt() - totalFee
+			: undefined;
+
+	let parsedSendAmountAfterFee: number | undefined;
+	$: parsedSendAmountAfterFee = nonNullish(sendAmountAfterFee)
+		? formatTokenBigintToNumber({
+				value: sendAmountAfterFee,
+				unitName: $destinationToken.decimals,
```

Comment:
nitpick: in a similar usage we created another derived inside the context for the decimals directly

Reply:
Good tip, thanks! I'll check it 👍